ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js

```diff
@@ -0,0 +1,266 @@
+  const trackEvent = useContext(MetaMetricsContext);
+  const mmiActions = mmiActionsFactory();
+
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const connectRequests = useSelector(
+    (state) => state.metamask.institutionalFeatures?.connectRequests,
+  );
+  const complianceActivated = useSelector((state) =>
+    Boolean(state.metamask.institutionalFeatures?.complianceProjectId),
+  );
+  const [showMore, setShowMore] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [connectError, setConnectError] = useState('');
+
+  const handleConnectError = (e) => {
+    let errorMessage = e.message;
+
+    if (!errorMessage) {
+      errorMessage = 'Connection error';
+    }
+
+    setConnectError(errorMessage);
+    setIsLoading(false);
+  };
+
+  const renderSelectedToken = () => {
+    const connectRequest = connectRequests ? connectRequests[0] : undefined;
+
+    return (
+      <Box
+        paddingTop={2}
+        paddingBottom={2}
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.ROW}
+        alignItems={AlignItems.center}
+      >
+        <Text>
+          {showMore && connectRequest?.token
+            ? connectRequest?.token
+            : `...${connectRequest?.token.slice(-9)}`}
+        </Text>
+        {!showMore && (
+          <Box paddingLeft={2}>
+            <a
+              rel="noopener noreferrer"
+              onClick={() => {
+                setShowMore(true);
+              }}
+            >
+              {t('ShowMore')}
+            </a>
+          </Box>
+        )}
+      </Box>
+    );
+  };
+
+  const connectRequest = connectRequests ? connectRequests[0] : undefined;
+
+  if (!connectRequest) {
+    history.push(mostRecentOverviewPage);
+    return null;
+  }
+
+  trackEvent({
+    category: 'MMI',
+    event: 'Custodian onboarding',
+    properties: {
+      actions: 'Custodian RPC request',
+      custodian: connectRequest.custodian,
+      apiUrl: connectRequest.apiUrl,
+    },
+  });
+
+  let custodianLabel = '';
+
+  if (
+    connectRequest.labels &&
+    connectRequest.labels.some((label) => label.key === 'service')
+  ) {
+    custodianLabel = connectRequest.labels.find(
+      (label) => label.key === 'service',
+    ).value;
+  }
+
+  return (
+    <Box className="page-container">
+      <Box className="page-container__header">
+        <Text className="page-container__title">{t('custodianAccount')}</Text>
+        <Text className="page-container__subtitle">
+          {t('mmiAddToken', [connectRequest.origin])}
+        </Text>
+      </Box>
+      <Box padding={4} className="page-container__content">
+        {custodianLabel && (
+          <>
+            <Text padding={4} color={TextColor.textDefault}>
+              {t('custodian')}
+            </Text>
+            <Label
+              marginRight={4}
+              marginLeft={4}
+              color={TextColor.textAlternative}
+              fontSize="14"
```

Comment:
This prop won't work. I would suggest using the default Label size or if it must be changed use `variant` prop and the `TextVariant` enum from design-system helpers. Reminder we use responsive typography so the font size will change at different screen sizes are you sure you haven't viewed it at a certain size?